components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/MetadataNode.java

```diff
@@ -0,0 +1,133 @@
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_LAST_UPDATED;
+
+/**
+ * Wraps an {@code OrientVertex} to provide a simpler API for working with stored component and asset metadata.
+ */
+public abstract class MetadataNode
+{
+  protected final OrientVertex vertex;
+
+  protected final NestedAttributesMap attributes;
+
+  MetadataNode(OrientVertex vertex) {
+    this.vertex = checkNotNull(vertex);
+    Map<String, Object> backing = vertex.getProperty(P_ATTRIBUTES);
+    this.attributes = new NestedAttributesMap(P_ATTRIBUTES, backing);
+  }
+
+  /**
+   * Gets the underlying vertex.
+   */
+  public OrientVertex vertex() {
+    return vertex;
+  }
+
+  /**
+   * Gets the identity of the underlying record.
+   */
+  public ORID identity() {
+    return vertex.getRecord().getIdentity();
+  }
+
+  /**
+   * Tells whether the underlying record is new.
+   */
+  public boolean isNew() {
+    return identity().isNew();
+  }
+
+  /**
+   * Gets all top-level property names for this node.
+   */
+  public Set<String> propertyNames() {
+    return vertex.getPropertyKeys();
+  }
+
+  /**
+   * Gets the "attributes" property of this node, a map of maps that is possibly empty, but never {@code null}.
+   */
+  public NestedAttributesMap attributes() {
+    return attributes;
+  }
+
+  /**
+   * Gets the "last_updated" property of the node, or {@code null} if it has never been saved.
+   */
+  @Nullable
+  public DateTime lastUpdated() {
+    return get(P_LAST_UPDATED, DateTime.class);
+  }
+
+  public DateTime requireLastUpdated() {
+    return require(P_LAST_UPDATED, DateTime.class);
+  }
+
+  /**
+   * Gets a property value as a string, or {@code null} if undefined.
+   */
+  public String get(final String name) {
+    return get(name, String.class);
+  }
+
+  /**
+   * Gets a property value coerced to the given type, or {@code null} if undefined.
+   */
+  @Nullable
+  @SuppressWarnings("unchecked")
+  public <T> T get(final String name, final Class<T> type) {
+    checkNotNull(name);
+    checkNotNull(type);
+
+    Object value = vertex.getProperty(name);
+    if (value != null) {
+      // TODO: PropertyEditor coercion? (see also, AttributesMap#coerce())
+      if (type == DateTime.class) {
+        return (T) new DateTime(value);
+      }
+      return (T) type.cast(value);
+    }
+    return null;
+  }
+
+  public String require(final String name) {
+    return require(name, String.class);
+  }
+
+  public <T> T require(final String name, final Class<T> type) {
+    T value = get(name, type);
+    checkState(value != null, "Missing property: " + name);
+    return value;
+  }
+
+  /**
+   * Sets a property to a given value, or {@code null} to remove it.
+   */
+  public void set(final String name, final @Nullable Object value) {
+    checkNotNull(name);
+    if (value == null) {
+      vertex.removeProperty(name);
+    }
+    else if (value instanceof DateTime) {
+      vertex.setProperty(name, ((DateTime) value).toDate());
```

Comment:
it does appear there's a way to register a custom serializer/deserializer, but what detail i've found has been incomplete. 

* http://jezzper.com/jezzper/discussions.nsf/0/4B3FB2D9D145E7C9C12579AE0048F722
* https://github.com/dozed/orientdb-test/blob/master/src/main/java/org/noorg/orientdb/test/Database.java#L63

I'll see what I can figure out. Also pinged luca.